### PR TITLE
Add AncestryDNA

### PR DESCRIPTION
**Pull request by madprime**

### CODE REVIEW

File: data_retrieval/ancestry_dna.py [hunk 1]
```diff
@@ -0,0 +1,414 @@
+r"""
+23andMe genotyping data extraction.
```
Comment (beaugunderson) [inline data_retrieval/ancestry_dna.py]: comment still refers to 23andMe

File: data_retrieval/ancestry_dna.py [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+
+Copyright (C) 2014 PersonalGenomes.org
+
+This software is shared under the "MIT License" license (aka "Expat License"),
+see LICENSE.TXT for full license text.
+
+May be used on the command line. For example, the following command:
+
+    python -m data_retrieval.twentythree_and_me <23andme_token> \
```
Comment (beaugunderson) [inline data_retrieval/ancestry_dna.py]: 23andMe -> Ancestry

File: data_retrieval/ancestry_dna.py [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+        vcf_data['ANCESTRYDNA_DATA'] = genotype_indexed
+        output_line = '\t'.join([vcf_data[x] for x in VCF_FIELDS])
+        output.write(output_line + '\n')
+
+    return output
+
+
+def clean_raw_ancestrydna(input_filepath, sentry=None, username=None):
+    """
+    Create clean file in 23andme format from downloaded version
```
Comment (beaugunderson) [inline data_retrieval/ancestry_dna.py]: 23andMe -> Ancestry

File: data_retrieval/ancestry_dna.py [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+
+    filename_base = 'AncestryDNA-genotyping'
+
+    raw_ancestrydna, genome_sex = clean_raw_ancestrydna(
+        input_file, sentry, username)
+    raw_ancestrydna.seek(0)
+    vcf_ancestrydna_unsorted = vcf_from_raw_ancestrydna(
+        raw_ancestrydna, genome_sex)
+
+    # Save raw 23andMe genotyping to temp file.
```
Comment (beaugunderson) [inline data_retrieval/ancestry_dna.py]: 23andMe -> Ancestry

File: data_retrieval/ancestry_dna.py [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+        temp_files.append({
+            'temp_filename': raw_filename,
+            'tempdir': tempdir,
+            'metadata': {
+                'description': "AncestryDNA full genotyping data, original format",
+                'tags': ['AncestryDNA', 'genotyping'],
+            },
+        })
+
+    # Save VCF 23andMe genotyping to temp file.
```
Comment (beaugunderson) [inline data_retrieval/ancestry_dna.py]: 23andMe -> Ancestry

**Comment by beaugunderson** [comment]
:+1: after replacing references to 23andMe with Ancestry :)

**Comment by madprime** [comment]
Haha, whoops! I'll note for the record that it _was_ in the end substantially edited from the copy-pasted original.

**Pull request merged**